src/Http/Response.php

```diff
@@ -2233,6 +2233,30 @@ public function withFile($path, array $options = [])
     }
 
     /**
+     * Convenience method to set a string into the response body
+     *
+     * @param string $string The string to be sent
+     * @return static
+     */
+    public function withStringBody($string)
+    {
+        $new = clone $this;
+        $body = $new->getBody();
+
+        if (is_string($string) === false) {
+            if ($body->isSeekable()) {
+                $body->rewind();
+            }
+            $body->write('');
+
+            return $new;
+        }
+        $body->write($string);
```

Comment:
You could replace the entire stream with a new instance as this method is explicitly replacing the body.